src/main/java/com/yelp/nrtsearch/server/config/NrtsearchConfig.java

```diff
@@ -434,4 +434,25 @@ private static List<String> getPluginSearchPath(Object o) {
     }
     return paths;
   }
+
+  @SuppressWarnings("unchecked")
+  public Map<String, Map<String, Object>> getIngestionPluginConfigs() {
+    try {
+      Object raw = configReader.get("pluginConfigs.ingestion", obj -> obj);
+      if (raw instanceof Map<?, ?> outerMap) {
+        Map<String, Map<String, Object>> result = new HashMap<>();
+        for (Map.Entry<?, ?> entry : outerMap.entrySet()) {
+          if (entry.getKey() instanceof String pluginName
+              && entry.getValue() instanceof Map<?, ?> pluginConfig) {
+            result.put(pluginName, (Map<String, Object>) pluginConfig);
+          }
+        }
+        return result;
```

Comment:
You could simply this code a bit using a jackson annotated class. You can load it like https://github.com/Yelp/nrtsearch/blob/main/src/main/java/com/yelp/nrtsearch/server/config/ThreadPoolConfiguration.java#L173

Reply:
done